### ISSUE TITLE
Docker HEALTHCHECK

### ISSUE THREAD

**Issue by jennydaman** [FNNDSC/ChRIS_ultron_backEnd#318]
Use `HEALTHCHECK`, can be something simple like

```bash
curl http://localhost:8000/api/v1/users/
```

Add documentation to describe more advanced health checking. For example, providing a user login for the healthcheck allows us to do

```bash
curl -u chris:chris1234 http://localhost:8000/api/v1/
```

This endpoint will cause the backend to do an SQL query, so that the health check is more informative.

Healthcheck in production cases should be customized because `DJANGO_ALLOWED_HOSTS` might disallow `localhost:8000`

**Comment by AfiMaameDufie** [comment]
Hi @jennydaman 
I am Afi, an Outreachy applicant. Can you assign this to me to work on if its available?

**Comment by beingnile** [comment]
Hello @jennydaman, My name is Nile Okomo(GMT + 3), an outreachy applicant interested in being assigned this issue.
Could I possibly work on it with @AfiMaameDufie

**Comment by jennydaman** [comment]
@AfiMaameDufie please go ahead. If you are open to collaborate, please reach out to @beingnile and work together with them.

Ideally, a healthcheck endpoint should be able to be queried anonymously, so I think it's a better idea to poll `http://localhost:8000/api/v1/users/`. As a stretch goal, you could implement a new endpoint `/api/v1/health/` which verifies database connection as well before responding 200 OK.

**Comment by StrangeJay** [comment]
Hi @jennydaman, i'm an outreachy applicant and i'd like to be assigned this work. I see it was assigned last year, but it's still an open issue so i'm guessing it wasn't done.

**Comment by jennydaman** [comment]
Yup please go ahead!

**Comment by StrangeJay** [comment]
Okay, i'd like more information on what's expected of me while resolving the issue. I spun up minichris locally and implemented a health check on it, but i noticed minichris is from a different repo, am i meant to spin up the dev env from this backend repo and make changes to it??? 

Am i making changes to the docker-compose file here or just documenting health checking on the readme.md???

**Comment by tr-sangale** [comment]
hi @jennydaman , I'm an outreachy intern looking forward to contribute to ChRIS, can i please get assigned to the issue?

**Comment by jennydaman** [comment]
@StrangeJay `HEALTHCHECK` is a directive which can be added to Dockerfile

https://docs.docker.com/engine/reference/builder/#healthcheck

ChRIS_ultron_backEnd is the source code for our backend, whereas miniChRIS-docker is an example of how to deploy ChRIS_ultron_backEnd. So to add the healthcheck, I think the best place to put it would be inside the `Dockerfile` of ChRIS_ultron_backEnd, and then once the PR is merged, update miniChRIS-docker.

**Comment by StrangeJay** [comment]
Okay, on it... Getting a lot of errors trying to start up the backend, but i'm working through them.
